### PR TITLE
docs: enhance wandb.Image docstrings for better normalization documentation

### PR DESCRIPTION
**Pull request by mdlinville**
## Summary

This PR addresses [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016) and the related GitHub issue [wandb/wandb#7377](https://github.com/wandb/wandb/issues/7377) by enhancing the docstrings in the `wandb.Image` implementation to provide better auto-generated documentation for image normalization behavior.

This is part of a three-part project. The other two PRs are:

- New example Colab notebook: https://github.com/wandb/examples/pull/609
- Docs change: https://github.com/wandb/docs/pull/1559

### Enhanced Docstrings (`wandb/sdk/data_types/image.py`)

1. **`_guess_and_rescale_to_0_255` function**:
   - Added comprehensive docstring explaining the normalization algorithm
   - Detailed explanation of supported input ranges: [0,1], [-1,1], and other ranges
   - Clear documentation of arguments, return values, and behavior
   - Examples of when normalization is applied vs avoided

2. **`Image.__init__` method**:
   - Added "Normalization behavior" section to the docstring
   - Detailed explanation of when normalization occurs (PyTorch tensors, NumPy arrays) vs when it doesn't (PIL Images, file paths)
   - Examples demonstrating normalization effects for different input ranges
   - Instructions on how to avoid normalization using PIL Images or the `normalize` parameter

## Problem Solved

The original issue was that users found the existing documentation about image normalization to be "cryptic" and unclear. Users were wasting computation time due to misunderstandings about:
- When normalization occurs
- How the normalization algorithm works
- What input ranges are supported
- How to avoid unwanted normalization

These enhanced docstrings will improve the auto-generated API documentation and provide clearer guidance to users.

## Related PRs

- Companion PR in `wandb/docs` repository: [wandb/docs#1559](https://github.com/wandb/docs/pull/1559)

## JIRA Reference

- [DOCS-1016](https://wandb.atlassian.net/browse/DOCS-1016): "Productboard: [Docs] Better explanation for the section: when logging a PyTorch tensor as a PIL Image"

[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCS-1016]: https://wandb.atlassian.net/browse/DOCS-1016?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mdlinville** [comment]
**Update**: Companion PR in `wandb/docs` repository: [wandb/docs#1559](https://github.com/wandb/docs/pull/1559)

File: wandb/sdk/data_types/image.py [hunk 1]
```diff
@@ -59,17 +59,34 @@ def _warn_on_invalid_data_range(
 def _guess_and_rescale_to_0_255(data: "np.ndarray") -> "np.ndarray":
```
Comment (timoffex) [inline wandb/sdk/data_types/image.py:59]: This function is private and shouldn't appear in our documentation; why update its docstring here? I prefer the original for conciseness and developer context. The new version is too verbose and adds irrelevant context about PyTorch, PIL and the `normalize` parameter to the `wandb.Image` constructor.

File: wandb/sdk/data_types/image.py [hunk 2]
```diff
@@ -237,6 +264,40 @@ def __init__(
...
+        import torch
+        import wandb
+        from PIL import Image as PILImage
+
+        with wandb.init() as run:
+            # Example 1: [0, 1] range tensor - values will be multiplied by 255
+            tensor_0_1 = torch.rand(3, 64, 64)  # Random values between 0 and 1
+            image1 = wandb.Image(tensor_0_1, caption="Normalized from [0,1] range")
+
+            # Example 2: [-1, 1] range tensor - values will be rescaled
```
Comment (timoffex) [inline wandb/sdk/data_types/image.py]: Not necessarily. If the values all happen to be nonnegative, this will be rescaled using the [0, 1] rules.

File: wandb/sdk/data_types/image.py [hunk 3]
```diff
@@ -169,8 +186,18 @@ def __init__(
                 If the values are not in the range [0, 255] or all values are in the range [0, 1],
                 the image pixel values will be normalized to the range [0, 255]
                 unless `normalize` is set to False.
```
Comment (timoffex) [inline wandb/sdk/data_types/image.py:163]: Should this still be here?

**Comment by mdlinville** [comment]
@timoffex PTAL, I think that your feedback in this PR is addressed, but let me know if there is more to do. Note that there is a related Docs PR and a related Examples PR to add a demo notebook.

File: wandb/sdk/data_types/image.py (hunk 1 above)
Comment (mdlinville) [reply wandb/sdk/data_types/image.py:59]: I'm a little confused about this feedback:

1. I am not sure why the _guess_and_rescale_to_0_255 function is showing up. @ngrayluna can you take a look? Is there some magic that prevents things starting with an underscore from showing up? Though, I think that this function doesn't show up in the docs, so maybe this is fine. https://docs.wandb.ai/ref/python/sdk/data-types/image/ is the currently published reference, without the changes in this PR.
2. Preferring more verbose: Is this about lines 66-72 vs lines 60-64 ? Would you like me to revert the change here?

File: wandb/sdk/data_types/image.py (hunk 3 above)
Comment (mdlinville) [reply wandb/sdk/data_types/image.py:163]: I'm not sure what you are specifically referring to. It looks to me like this is missing the -1-1 range case. Do you mean that since we have lines 62-64, the detail in 161-3 should be removed?

**Comment by dmitryduev** [comment]
@mdlinville is this still relevant or we can close?
